docs/proposals/koordlet/20250425-heterogeneous-GPU-device-reporting.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Heterogeneous GPU device reporting
+authors:
+  - "@ZhuZhezz"
+reviewers:
+  - "@zqzten"
+  - "@ZiMengSheng"
+  - "@hormes"
+creation-date: 2025-04-25
+last-updated: 2025-04-25
+status: provisional
+---
+
+# Heterogeneous GPU device reporting
+
+## Summary
+
+This proposal introduces a heterogeneous GPU device reporting mechanism to koordlet, particularly for GPU devices with dynamic virtualization. The main goal is to decouple device awareness from Koordlet, maitain a clean Koordlet architecture and facilitate integration for third-party vendors.
+
+## Motivation
+For now, Koordlet is able to report NVIDIA GPU from NVML. But for third-party GPU devices, there is no standard way. If third-party GPU devices are integrated into Koordlet, it will introduce vendor-specific logic and SDK, such as DCML(Ascend Card) and IXML(Iluvatar Card). Meanwhile, we need to address the cgo dependencies issue. This will make Koordlet hard to maintain and more complex.
+
+This PR proposes a mechanism to report devices, and makes Koordlet more extensible and maintainable.
+
+### Goals
+
+- Introduce a mechanism for reporting heterogeneous GPU devices.
+- Allow Koordlet integrates third-party GPU devices quickly.
+- Keep Koordlet architecture clean.
+
+## Proposal
+
+### Architecture
+![image](/docs/images/GPU-device-reporting.svg)
+
+### Implementation Details
+
+- Thirty-party vendors write [device infos](#Device-Infos-Formate-In-File) into directory `/etc/kubernetes/koordlet/device-infos/`.
+- Koordlet will read the device infos from the directory and report them to apiserver.
+  - The topology infos from directory will be synced to the annotation of the Device.
+```
+apiVersion: scheduling.koordinator.sh/v1alpha1
+kind: Device
+metadata:
+  annotations:
+    # The topology from device infos will be synced to the annotation of the Device.
+    scheduling.koordinator.sh/gpu-partitions: |
+      {
+        "1": [
+                {
+                  "minors": [
+                      0
+                  ],
+                  "gpuLinkType": "NVLink",
+                  "ringBusBandwidth": "400Gi",
+                  "allocationScore": "1"
+                }
+        ],
+        "4": [
+                {
+                    "minors": [
+                       0,1,2,3 
+                    ],
+                    # This is the topology of the 910B.
+                    # https://www.hiascend.com/doc_center/source/zh/mindx-dl/50rc1/ref/dl_affinity_0002.html
+                    "gpuLinkType": "HCCS",
+                    "allocationScore": "1"
+                }
+        ]
+      }
+  labels:
```

Comment:
Vendor and model shall be labelled.
new: `node.koordinator.sh/gpu-vendor`
already exists: `node.koordinator.sh/gpu-model`